doc/rtd/explanation/about-cloud-config.rst

```diff
@@ -0,0 +1,167 @@
+
+   #cloud-config
+   keyboard:
+     layout: us
+     model: pc105
+     variant: nodeadkeys
+     options: compose:rwin
+
+A full list of modules can be found :ref:`on our modules page <modules>`. This
+list also shows the valid schema keys for every module, and YAML examples.
+
+Module ordering
+---------------
+
+The order of the ``cloud-config`` keys is unimportant and modules can be
+written in any order.
+
+.. note::
+   The :ref:`Users and Groups <mod_cc_users_groups>` module is a special case
+   where ordering matters if you want to add users to a group list.
+
+Checking your cloud-config file
+===============================
+
+Once you have constructed your cloud-config file, you can check it against
+the :ref:`cloud-config validation tool <check_user_data_cloud_config>`. This
+tool tests the YAML in your file against the cloud-init schema and will warn
+you of any errors.
+
+Example cloud-config file
+=========================
+
+The following code is an example of a complete user data cloud-config file.
+The :ref:`cloud-config example library <yaml_examples>` contains further
+examples that can be copy/pasted and adapted to your needs.
+
+.. code-block:: yaml
+
+   #cloud-config
+
+   # Basic system setup
+   hostname: example-host
+   fqdn: example-host.example.com
+
+   # Configure storage
+   storage:
+     files:
+       - path: /etc/example_file.txt
+         content: |
+           Some text to be stored in the file
+       - path: /etc/example_script.txt
+         content: |
+           #!/bin/bash
+           echo "Some text to be run in the script"
+
+   # User setup configuration
+   users:
+     - name: exampleuser
+       gecos: Example User
+       sudo: ['ALL=(ALL) NOPASSWD:ALL']
+       groups: sudo
+       home: /home/exampleuser
+       shell: /bin/bash
+       ssh_authorized_keys:
+         - ssh-rsa AAAAB3...restofpublickey user@host
+
+   # Change passwords using chpasswd
+   chpasswd:
+     exampleuser: terriblepassword12345
+
+   # Package management
+   package_update: true
+   package_upgrade: true
+   packages:
+     - git
+     - nginx
+     - python3
+
+   # Commands to run at the end of the cloud-init process
+   runcmd:
+     - echo "Hello, world!" > /etc/motd
+     - systemctl restart nginx
+     - mkdir -p /var/www/html
+     - echo "<html><body><h1>Welcome to the party, pal!</h1></body></html>" > /var/www/html/index.html
+
+   # Write files to the instance
+   write_files:
+     - path: /etc/example_config.conf
+       content: |
+         [example-config]
+         key=value
+     - path: /etc/motd
+       content: |
+         Some text that will appear in your MOTD!
+
+   # Final message, shown after cloud-init completes
+   final_message: "The system is up, after $UPTIME seconds"
+
+   # Reboot the instance after configuration
+   power_state:
+     mode: reboot
+     message: Rebooting after initial setup
+     timeout: 30
+     condition: True
+
+Ubuntu installer cloud-config
```

Comment:
I don't think that this section makes sense in cloud-init's documentation about cloud-config. Cloud-init doesn't do anything special when autoinstall is included, it is just ignored (and subiquity does whatever it does with it later).

Reply:
A counterpoint here is that we are getting lots of questions in IRC in cloud-init about this and a lot of keyword searches `past 30 days: autoinstall (1 result) 136 searches`.   I'd rather have a link handy in our docs that says, not our deal so that hopefully it is discoverable before people ask in #cloud-init channel in IRC.

Reply:
> a lot of keyword searches past 30 days: autoinstall (1 result) 136 searches. I'd rather have a link handy in our docs that says, not our deal so that hopefully it is discoverable before people ask in #cloud-init channel in IRC.

Why is is this page the right place for it? There are many things that one could say about cloud-config, but there is value in keeping it short and on-topic without cluttering it with random details. Also, go search autoinstall in the docs. [I get exactly one hit](https://docs.cloud-init.io/en/latest/reference/faq.html#autoinstall-preruncmd-postruncmd). We don't need to say "not our deal" in two places, so I'd rather not add another unless we're planning on removing the existing one - and if we're going to relocate it lets put it where it belongs rather than from one section to the next. Your points make it sound like it is a frequently asked question, which is exactly the purpose of the page that that section is currently on.

Reply:
I agree that this doesn't belong here. I made a similar comment on an earlier version of this PR and would prefer to keep this info buried in the FAQ, but FAQs don't fit nicely into diataxis and are seen by some as an indication of bad documentation.

Could we keep it in the FAQ for now and perhaps grow a "Projects that use cloud-init" or similar page that point to other well known (and not just Canonical) projects that use cloud-init along with important things to be aware of?